### PR TITLE
update defaults for upnp and mdns, closes #664

### LINKED ISSUE #664: rationalize defaults of upnp & mdns

### PR DESCRIPTION
**Pull request by zippy**
for hcdev mode mdns defaults to true and upnp to false
for hcd the opposite

N.B. changed flag in `hcdev` from "-no-nat-upnp" -> "-upnp"

### CODE REVIEW

**Comment by thedavidmeister** [comment]
`The command "TEST_FLAGS='-v -coverprofile=$(pkg_path)/coverage.txt -covermode=atomic' make -e test" exited with 2.` in travis

**Comment by thedavidmeister** [comment]
definitely want to sweep the docs after this goes in as i'm sure there's a few references to the old flags scattered around

**Comment by zippy** [comment]
Re Docs: Absolutely.  Note that travis gets very grumpy about some timing issues in the tests.  I haven't been able to figure them out so I just re-run the tests till they go green.  The are all green on my machine.  I know this isn't optimal, but that's where we are right now...  sigh.

File: cmd/hcdev/hcdev.go [hunk 1]
```diff
@@ -128,15 +128,15 @@ func setupApp() (app *cli.App) {
...
-			Usage:       "whether to use mdns for local peer discovery",
+			Usage:       "whether to use mdns for local peer discovery (default = true)",
 			Destination: &mdns,
 		},
 		cli.BoolFlag{
-			Name:        "no-nat-upnp",
-			Usage:       "whether to stop hcdev from creating a port mapping through NAT via UPnP",
-			Destination: &nonatupnp,
+			Name:        "upnp",
+			Usage:       "whether to use UPnP for creating a NAT port mapping (default true)",
```
Comment (thedavidmeister) [inline cmd/hcdev/hcdev.go]: description of this PR says:

> for hcdev mode mdns defaults to true and upnp to false

**Comment by thedavidmeister** [comment]
@zippy random fails are why timing should be avoided in tests yeah

it starts innocent enough when 1/1000 test runs fail, but after you have 1000 test cases it becomes very hard to get a clean run :/

i've had this problem with selenium testing before... i don't use selenium any more :P

File: cmd/hcdev/hcdev.go (hunk 1 above)
Comment (zippy) [reply cmd/hcdev/hcdev.go]: duh! fixed.

**Comment by Connoropolous** [comment]
wooohoo yes :)

**Comment by thedavidmeister** [comment]
docs are updated to match logic and tests pass so i'm happy with this

followup issue at https://github.com/holochain/holochain-proto/issues/669

**Pull request merged**